transaction/std/src/transaction_builder.rs

```diff
@@ -143,6 +159,26 @@ impl<FPR: FogPubkeyResolver> TransactionBuilder<FPR> {
         recipient: &PublicAddress,
         rng: &mut RNG,
     ) -> Result<(TxOut, TxOutConfirmationNumber), TxBuilderError> {
+        let context = self.add_output_with_context(value, recipient, rng)?;
+        Ok((context.tx_out, context.confirmation))
+    }
+
+    /// Add a non-change output to the transaction.
+    ///
+    /// If a sender memo credential has been set, this will create an
+    /// authenticated sender memo for the TxOut. Otherwise the memo will be
+    /// unused.
+    ///
+    /// # Arguments
+    /// * `value` - The value of this output, in picoMOB.
+    /// * `recipient` - The recipient's public address
+    /// * `rng` - RNG used to generate blinding for commitment
+    pub fn add_output_with_context<RNG: CryptoRng + RngCore>(
+        &mut self,
+        value: u64,
+        recipient: &PublicAddress,
+        rng: &mut RNG,
+    ) -> Result<TxOutContext, TxBuilderError> {
```

Comment:
I think this changes the transaction interface a little too much without much tangible benefit given the shared secret is possible to derive elsewhere. The ffis already compute the secret in this way here - a similar approach should probably be used for the memo decryption or if really needed to get the shared secret https://github.com/mobilecoinfoundation/mobilecoin/blob/9b08d3c0a6e5f4b866a8734e7c10274e404c056b/android-bindings/src/bindings.rs#L377

Reply:
@iamalwaysuncomfortable that's not correct, the recipient of a TxOut is able to reconstruct the TxOut shared secret, but the sender is not, unless we expose either the tx out shared secret, or the tx private key, from the transaction builder. So they actually do need this or something like it if they want to encrypt additional messages using the shared secret.

Reply:
Good clarification!